falcon/request.py

```diff
@@ -358,10 +358,23 @@ class Request(object):
             accessed.)
         range_unit (str): Unit of the range parsed from the value of the
             Range header, or ``None`` if the header is missing
-        if_match (str): Value of the If-Match header, or ``None`` if the
-            header is missing.
-        if_none_match (str): Value of the If-None-Match header, or ``None``
-            if the header is missing.
+        if_match (list): A list containing all the etags in the If-Match
+            header.
```

Comment:
Regardless of the chosen design, we need to talk more about the entities within the list, i.e. strings with the _weak_ attribute, or w/e we choose.